course5/homework06/simple_string.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <iostream>
+#include <memory>
+
+namespace simple_string
+{
+
+class String
+{
+public:
+    String();
+    String(const char*);
+    String(const String& src);
+    String(String&& src);
+    String& operator=(const String& src);
+    String& operator=(String&& src);
+    String& operator=(const char * src);
+    String operator+(const char * src);
+    String operator+(const String& src);
+    String& operator+=(const char * src);
+    String& operator+=(const String& src);
+
+    char operator[](const int) const;
+    char& operator[](const int);
+
+    void Print() const;
+    int Length() const;
+    const char *CStr() const;
+
+private:
+    std::unique_ptr<char[]> buffer;
+    int size = 0;
+
+}; // String
+
+} // namespace
+
+std::ostream& operator<<(std::ostream&, const simple_string::String&);
```

Comment:
This really should be inside of the namespace. If you struggle with the linking errors, then the underlying problem should be solved.